custom/icds/tasks/sms.py

```diff
@@ -17,21 +18,31 @@
 if settings.SERVER_ENVIRONMENT in settings.ICDS_ENVS:
     @periodic_task(run_every=crontab(day_of_month='2', minute=0, hour=0))
     def send_monthly_sms_report():
-        start_date = date.today().replace(day=1) - relativedelta(months=1)
-        _, last_day = calendar.monthrange(start_date.year, start_date.month)
-        end_date = start_date.replace(day=last_day)
-        filename = call_command('get_icds_sms_usage', 'icds-cas', str(start_date), str(end_date))
-        with open(filename, 'rb') as f:
-            cached_download = expose_cached_download(f.read(), expiry=24 * 60 * 60, file_extension='xlsx')
-        link = "%s%s" % (web.get_url_base(), reverse('retrieve_download',
-                                                     kwargs={'download_id': cached_download.download_id}))
-        message = """
-        Hi,
-        Please download the sms report for last month at {link}.
-        The report is available only till midnight today.
-        """.format(link=link)
         subject = _('Monthy SMS report')
         recipients = ['mkangia@dimagi.com', 'ayogi@dimagi.com', 'adhaar.kaul@gmail.com',
                       'umra.liaqat@gmail.com']
-        send_html_email_async.delay(subject, recipients, message,
-                                    email_from=settings.DEFAULT_FROM_EMAIL)
+        try:
+            start_date = date.today().replace(day=1) - relativedelta(months=1)
+            first_day, last_day = calendar.monthrange(start_date.year, start_date.month)
+            end_date = start_date.replace(day=last_day)
+            filename = call_command('get_icds_sms_usage', 'icds-cas', str(start_date), str(end_date))
+            with open(filename, 'rb') as f:
+                cached_download = expose_cached_download(f.read(), expiry=24 * 60 * 60, file_extension='xlsx')
+            link = "%s%s" % (web.get_url_base(), reverse('retrieve_download',
+                                                         kwargs={'download_id': cached_download.download_id}))
+            message = _("""
+            Hi,
+            Please download the sms report for last month at {link}.
+            The report is available only till midnight today.
+            """).format(link=link)
+            send_html_email_async.delay(subject, recipients, message,
+                                        email_from=settings.DEFAULT_FROM_EMAIL)
+        except Exception as e:
+            message = _("""
+                Hi,
+                Could not generate the montly SMS report for ICDS.
+                The error has been notified. Please report as an issue for quick followup
+            """)
+            send_html_email_async.delay(subject, recipients, message,
```

Comment:
Might it be useful to include a stack trace or anything?

Reply:
I suppose we should be able to find it in sentry based on the timestamp of the email.

Reply:
I think better to view it on sentry itself. Should not be hard to locate using the method name?
Not strongly against adding it here, just that most of the recipients would make sense out of it and will just scare them.

Reply:
good point